### PR TITLE
Add a static classification tagger for previews

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/EditorFeatures/Core.Wpf/Preview/PreviewStaticClassificationTaggerProvider.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Preview
+{
+    /// <summary>
+    /// This tagger assumes content of the buffer never get changed. 
+    /// and the buffer provides static classification information on the buffer content
+    /// through <see cref="PredefinedPreviewTaggerKeys.StaticClassificationSpansKey" /> in the buffer property bag
+    /// </summary>
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Preview/PreviewStaticClassificationTaggerProvider.cs]: i wouldn't mind a more fleshed out doc comment.

File: src/EditorFeatures/Core.Wpf/Preview/PreviewStaticClassificationTaggerProvider.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            event EventHandler<SnapshotSpanEventArgs> ITagger<IClassificationTag>.TagsChanged
+            {
+                add
+                {
+                }
+
+                remove
+                {
+                }
+            }
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core.Wpf/Preview/PreviewStaticClassificationTaggerProvider.cs]: feels really unnecessary to use 10 lines for a no-op event handler :)

Comment (sharwell) [reply src/EditorFeatures/Core.Wpf/Preview/PreviewStaticClassificationTaggerProvider.cs]: ➡️ Collapsed to one line

File: src/EditorFeatures/Core.Wpf/Preview/PreviewStaticClassificationTaggerProvider.cs (hunk 1 above)
Comment (sharwell) [reply src/EditorFeatures/Core.Wpf/Preview/PreviewStaticClassificationTaggerProvider.cs]: Will file a follow-up issue for this

Comment (sharwell) [reply src/EditorFeatures/Core.Wpf/Preview/PreviewStaticClassificationTaggerProvider.cs]: #31071

**Pull request merged**